src/accelerate/utils/operations.py

```diff
@@ -234,6 +234,8 @@ def find_batch_size(data):
     if isinstance(data, (tuple, list)):
         return find_batch_size(data[0])
     elif isinstance(data, Mapping):
+        if len(data) == 0:
+            raise ValueError(f"Can't find the batch size from empty {type(data)}.")
```

Comment:
If we're going to do this, let's cover all the bases then. Let's do this as the following then as the first two lines in the func:

```python
if isinstance(data, (tuple, list, Mapping)) and (len(data) == 0):
    raise ValueError(f"Cannot find the batch size from empty {type(data)}.")
```

Reply:
Agree, for some reason I just missed how the first branch works. Added this change and empty line to differ these two "if"s.